site/_guides/gatekeeper.md

```diff
@@ -0,0 +1,450 @@
+
+_* Note that the HTTPProxy is still marked invalid by Contour after creation because the service `s1` does not exist, but that's outside the scope of this guide._
+
+Finally, create a file called `httpproxy-retries-default.yml` containing the following YAML:
+
+```yaml
+apiVersion: projectcontour.io/v1
+kind: HTTPProxy
+metadata:
+  name: demo-retries
+  namespace: default
+spec:
+  virtualhost:
+    fqdn: default.retries.projectcontour.io
+  routes:
+    - conditions:
+        - prefix: /foo
+      services:
+        - name: s1
+          port: 80
+      retryPolicy:
+        count: 6
+```
+
+Remember that our `Constraint` was defined to apply only to the `my-namespace` namespace, so it should not block the creation of this proxy, even though it has a retry policy count outside the allowed range.
+
+Apply the YAML:
+
+```bash
+$ kubectl apply -f httpproxy-retries-default.yml
+```
+
+The `HTTPProxy` creates successfully.
+
+## Gatekeeper Audit
+
+We've seen how Gatekeeper constraints can enforce constraints when a user tries to create a new `HTTPProxy`. Now let's look at how constraints can be applied to pre-existing resources in the cluster.
+
+Gatekeeper has an audit functionality, that periodically (every `60s` by default) checks all existing resources against the relevant set of constraints. Any violations are reported in the `Constraint` custom resource's `status.violations` field. This allows an administrator to periodically review & correct any pre-existing misconfigurations, while not having to worry about breaking existing resources when rolling out a new or updated constraint.
+
+To try this out, let's revisit the previous example, and change our constraint to allow a maximum retry count of four.
+
+Edit `retry-count-range-constraint.yml` and change the `max` field to have a value of `4`. Save the file.
+
+Apply it to the cluster:
+
+```bash
+$ kubectl apply -f retry-count-range-constraint.yml
+```
+
+We know that the `demo-retries` proxy has a route with a `retryPolicy.count` of `5`. This should now be invalid according to the updated constraint.
+
+Wait up to `60s` for the next periodic audit to finish, then run:
+
+```bash
+$ kubectl describe httpproxyretrycountrange httpproxy-retry-count-range
+```
+
+You should see something like:
+
+```
+...
+Status:
+    ...
+    Violations:
+        Enforcement Action:  deny
+        Kind:                HTTPProxy
+        Message:             retry policy count must be less than/equal to 4
+        Name:                demo-retries
+        Namespace:           default
+```
+
+However, our `HTTPProxy` remains in the cluster and can continue to route requests, and the user can remediate the proxy to bring it inline with the policy on their own timeline.
+
+## Next steps
+
+Contour has a [growing library](https://github.com/projectcontour/contour/tree/master/examples/gatekeeper) of Gatekeeper constraint templates and constraints, for both **validations** and **policies**.
```

Comment:
I really like the idea of the project hosting a growing collection of templates and constraints. What I would suggest to make this more manageable is to separate each YAML object into a separate file. I think that your classification of validations and policies makes sense, so we can continue it, maybe like this:

```
002-validation-template-Foo-Name.yaml
002-validation-template-Bar-Name.yaml
003-validation-constraint-Foo.yaml
003-validation-constraint-Bar.yaml
004-policy-template-A.yaml
004-policy-template-B.yaml
```

This preserves the classification, but makes it a bit easier to pick and choose.

Reply:
I ended up creating two subdirectories, one for `validations` and one for `policies`, and then split things out to a resource per file, per your suggestion.